prow/rewriteConfig.go

```diff
@@ -0,0 +1,214 @@
+	i := 0
+	for _, runeValue := range line {
+		if runeValue == ' ' {
+			i++
+		} else {
+			break
+		}
+	}
+	return i
+}
+
+// Add element to the slice at the indexed position and move all the strings from
+// the index one position to the right.
+func appendAtIndex(original string, element string, index int) string {
+	prev := original[:index]
+	apres := original[index:]
+	return prev + element + apres
+}
+
+// Find index of master section in each branch split for the new branch and
+// get content from master section with mandatory "merges-blocked-needs-admin" part.
+func getContentOfNewBranch(branchContent string, newBranch string) (int, string) {
+	branchLines := strings.Split(branchContent, "\n")
+	if len(branchLines) < 2 {
+		return -1, ""
+	}
+
+	firstBranchLine := branchLines[1]
+	spacesForBranchNum := countLeadingSpace(firstBranchLine)
+
+	spacesForBranch := strings.Repeat(" ", spacesForBranchNum)
+	masterLine := spacesForBranch + "master:\n"
+	masterStart := strings.Index(branchContent, masterLine)
+
+	nonMasterContent := branchContent[masterStart+1:]
+
+	generalBranchRe := regexp.MustCompile("\n" + spacesForBranch + "\\S+:\n")
+	otherBranches := generalBranchRe.FindStringSubmatchIndex(nonMasterContent)
+
+	var masterContent string
+	if otherBranches == nil {
+		masterContent = branchContent[masterStart:]
+	} else {
+		masterContent = branchContent[masterStart:otherBranches[0]]
+	}
+
+	masterContent = strings.Join(strings.Split(masterContent, "\n")[1:], "\n")
+	contextRe := regexp.MustCompile("\\s+contexts:\n")
+	contextInd := contextRe.FindStringSubmatchIndex(masterContent)
+	if contextInd == nil {
+		masterContentSplit := strings.Split(masterContent, "\n")
+		masterContentLine := masterContentSplit[0]
+		spacesInMaster := countLeadingSpace(masterContentLine)
+
+		tab := spacesInMaster - spacesForBranchNum
+		statusCheckString := strings.Repeat(" ", spacesInMaster) + "required_status_checks:\n"
+		contextsString := strings.Repeat(" ", spacesInMaster+tab) + "contexts:\n"
+
+		if strings.Compare(masterContentSplit[len(masterContentSplit)-1], "") != 0 {
+			masterContent = masterContent + "\n"
+		}
+
+		masterContent = masterContent + statusCheckString + contextsString
+	}
+
+	contextInd = contextRe.FindStringSubmatchIndex(masterContent)
+	contextStart := contextInd[0] + 1
+	contextContent := masterContent[contextStart:]
+	if !strings.Contains(contextContent, "- \"merges-blocked-needs-admin\"") {
+		contextSpaces := countLeadingSpace(masterContent[contextStart:contextInd[1]])
+		adminLine := strings.Repeat(" ", contextSpaces) + "- \"merges-blocked-needs-admin\"\n"
+
+		if strings.Compare(string(masterContent[len(masterContent)-1]), "\n") != 0 {
+			masterContent = masterContent + "\n"
+		}
+
+		masterContent = masterContent + adminLine
+	}
+
+	withinMaster := spacesForBranch + newBranch + ":\n" + masterContent
+	return masterStart, withinMaster
+}
+
+func findRepo(source []byte, repos []string, newBranch string) string {
```

Comment:
Can you add comments to what are you looking for in this function?
for example line 118, why there is if statement?